pallet/crowdsale/src/lib.rs

```diff
@@ -445,9 +445,9 @@ pub mod pallet {
 			let who = ensure_signed(origin)?;
 
 			// update the sale status if the start block is met
-			SaleInfo::<T>::try_mutate(sale_id, |sale_info: &mut Option<SaleInformation<_, _>>| {
+			SaleInfo::<T>::try_mutate(sale_id, |sale_info| -> DispatchResult {
 				let Some(sale_info) = sale_info else {
-					return Err(Error::<T>::CrowdsaleNotFound);
+					return Err(Error::<T>::CrowdsaleNotFound.into());
 				};
```

Comment:
nit: - can apply the same everywhere we do this to reduce lines of code
```suggestion
				let sale_info = sale_info.as_mut().ok_or(Error::<T>::CrowdsaleNotFound)?;
```